### PR TITLE
fix(appeals): fix site-visit

### PR DESCRIPTION
**Pull request by AlexMoulin**
Fixes site visit schedule, prepopulates with exisiting values, saves data, and redirects correctly

## Type of change 🧩

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [x] I have performed a self-review of my own code
- [x] I have double checked this work does not include any hardcoded secrets or passwords
- [x] I have made corresponding changes to the documentation
- [x] I have provided details on how I have tested my code
- [x] I have referenced the ticket number above
- [x] I have provided a description of the ticket
- [x] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: appeals/web/src/server/appeals/appeal-details/site-visit/site-visit.controller.js [hunk 1]
```diff
@@ -38,6 +38,30 @@ const renderScheduleSiteVisit = async (request, response) => {
 			}
 		} = request;
 
+		// Nullish coalescing assignment https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_assignment
+		visitType ??= appealDetails.siteVisit.visitType;
```
Comment (james-hill-kc) [inline appeals/web/src/server/appeals/appeal-details/site-visit/site-visit.controller.js:42]: Does this mean existing values from the API response will overwrite values in the POST request body? If so I think it should be the other way around

File: appeals/web/src/server/views/appeals/appeal/schedule-site-visit.njk [hunk 2]
```diff
@@ -79,17 +79,17 @@
 				{
 					value: "unaccompanied",
 					text: "Unaccompanied",
-					checked: siteVisit.visitType === "unaccompanied"
+					checked: (siteVisit.visitType === "unaccompanied" or siteVisit.visitType === "Unaccompanied")
```
Comment (james-hill-kc) [inline appeals/web/src/server/views/appeals/appeal/schedule-site-visit.njk:82]: If this is to handle values from the API response, it should probably be done in the mapper instead

**Review by james-hill-kc — CHANGES_REQUESTED** [review]
Looks good! Just a couple of comments from me

File: appeals/web/src/server/appeals/appeal-details/site-visit/site-visit.controller.js (hunk 1 above)
Comment (AlexMoulin) [reply appeals/web/src/server/appeals/appeal-details/site-visit/site-visit.controller.js:42]: Yes but only if they are not defined.

File: appeals/web/src/server/views/appeals/appeal/schedule-site-visit.njk (hunk 2 above)
Comment (AlexMoulin) [reply appeals/web/src/server/views/appeals/appeal/schedule-site-visit.njk:82]: I think the best thing to do would be to fix the API so the Get (Unaccompanied) and Post (unaccompanied) are the same value formats. I'll create a ticket to address that and put a note to remove redundant code after it has been done. Would that be sufficient or should I build a temporary solution in the mapper too?

File: appeals/web/src/server/appeals/appeal-details/site-visit/site-visit.controller.js (hunk 1 above)
Comment (james-hill-kc) [reply appeals/web/src/server/appeals/appeal-details/site-visit/site-visit.controller.js:42]: Oh I see now, I was misunderstanding how nullish coalescing works

File: appeals/web/src/server/views/appeals/appeal/schedule-site-visit.njk (hunk 2 above)
Comment (james-hill-kc) [reply appeals/web/src/server/views/appeals/appeal/schedule-site-visit.njk:82]: I agree this should be fixed API side ideally. In principle though there is still mapping occurring directly in the view here, strictly speaking it should be moved into the mapper regardless of what the API values are

**Review by james-hill-kc — APPROVED** [review]
Looks good!

**Pull request merged**